sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -189,15 +191,47 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
-        if self.freq == "auto" and fh.freq is None:
-            # when freq cannot be interpreted from ForecastingHorizon
-            raise ValueError(
-                f"Error in {self.__class__.__name__}, "
-                f"could not interpret freq, "
-                f"try passing freq in model initialization"
-            )
-
-        self._freq = fh.freq if self.freq == "auto" else self.freq
+        # A. freq is given {use this}
+        # B. freq is auto
+        #     B1. freq is infered from fh {use this}
+        #     B2. freq is not infered from fh
+        #         B2.1. y is date-like {raise exception}
+        #         B2.2. y is not date-like
+        #             B2.2.1 equispaced integers {use diff in time}
+        #             B2.2.2 non-equispaced integers {raise exception}
+
+        if self.freq != "auto":
+            # A
+            self._freq = self.freq
+        else:
+            # B
+            if fh.freq:
+                # B1
+                self._freq = fh.freq
+            else:
+                # B2
+                if isinstance(y.index, (pandas.PeriodIndex, pandas.DatetimeIndex)):
```

Comment:
Question: is it the case that `pandas.PeriodIndex` always has `freq` attribute? If so, may be B2 should never arise if `y` has period index.

Can you please check this? If so, we can check for just DatetimeIndex. This is optional, and non-blocking.

Reply:
Yes, that case should work theoretically and I've checked, all the test cases are successful in that case as well.